arquillian-environment-reporter/arquillian-environment-reporter-impl/src/main/java/org/arquillian/environment/reporter/impl/ArquillianEnvironmentReporterLifecycleManager.java

```diff
@@ -0,0 +1,64 @@
+package org.arquillian.environment.reporter.impl;
+
+import org.arquillian.reporter.api.builder.Reporter;
+import org.arquillian.reporter.api.event.SectionEvent;
+import org.arquillian.reporter.api.event.TestSuiteConfigurationSection;
+import org.arquillian.reporter.api.model.report.ConfigurationReport;
+import org.jboss.arquillian.core.api.Event;
+import org.jboss.arquillian.core.api.annotation.Inject;
+import org.jboss.arquillian.core.api.annotation.Observes;
+import org.jboss.arquillian.test.spi.event.suite.BeforeSuite;
+
+import java.nio.charset.Charset;
+import java.util.TimeZone;
+
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.CHARSET;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.DOCKER;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.ENVIRONMENT_SECTION_NAME;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.JAVA_VERSION;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.OPERATIVE_SYSTEM;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.TEST_RUNNER;
+import static org.arquillian.environment.reporter.impl.ArquillianEnvironmentKeys.TIMEZONE;
+
+public class ArquillianEnvironmentReporterLifecycleManager {
```

Comment:
As for the name in other extensions:
arquillian-core-reporter - the name makes sense there as it manages the lifecycle of an arquillian test
core - well, this one is questionable, but I still believe that it makes sense as when the first event is observed then the report is created, then you check all events during the test and at the end, you create the report...

as for this one. you are right that there is no lifecycle that would be managed, but I understand the Alex's intention. 
@bartoszmajsak do you have any proposals? not only for this class but also for some convention...